lightning/src/chain/mod.rs

```diff
@@ -57,6 +57,81 @@ pub trait Listen {
 	fn block_disconnected(&self, header: &BlockHeader, height: u32);
 }
 
+/// The `Confirm` trait is used to notify when transactions have been confirmed on chain or
+/// unconfirmed during a chain reorganization.
+///
+/// Clients should use this interface instead of [`Listen`] if they receive transaction data only
+/// relevant to that identified by [`Filter`]. For instance, an Electrum client may implement
+/// [`Filter`] by subscribing to activity related to registered transactions and outputs. Upon
+/// notification, it would pass along the relevant transactions using this interface.
+///
+/// The intended use is as follows:
+/// - Call [`transactions_confirmed`] for any on-chain activity of interest.
+/// - Return such transactions from [`get_relevant_txids`] until they have enough confirmations to
+///   no longer be in danger of chain reorganization.
+/// - Call [`transaction_unconfirmed`] for any transaction returned by [`get_relevant_txids`] that
```

Comment:
If a transaction was confirmed, but was later reorged into a different/later block height, should transaction_confirmed be called again, or should transaction_unconfirmed be called first?

Reply:
Here, `transaction_unconfirmed` would need to be called first. I think this is addressed in the new "Order" section. Let me know if it doesn't clear things up, though, or if you have recommendations on how to better word it.